Server/Controllers/taskProgressController.js

```diff
@@ -0,0 +1,39 @@
+const {UserTaskProgress, Course, Task, Lecture} = require('../Models/index');
+const ApiError = require('../utils/ApiError');
+const asyncHandler = require('express-async-handler');
+
+exports.calculateAverageTaskProgress = asyncHandler(async (req, res, next) => {
+    const courseId = req.params.courseId;
+    const userId = req.user.id;
+    
+    const course = await Course.findByPk(courseId);
+    if (!course) {
+        return next(
+            new ApiError("No course was found with this id", 404)
+        )
+    }
+
+    const tasks_count = await Task.count({
+        include: [{
+            model: Lecture,
+            as: "lecture",
+            where: {
+                courseId,
+            },
+        }]
+    });
+    
+    if (!tasks_count) {
+        return res.status(200).json({"average_progress": tasks_count});
+    }
+    const tasks_completed_count = await UserTaskProgress.count({
+        where: {
+            is_finished: true,
+            userId,
+        },
+    });
+
+    const average_progress = (tasks_completed_count / tasks_count) * 100;
+
+    res.status(200).json({average_progress})
```

Comment:
Why don't you merge the `tasks_count` and `average_progress` together in one response
and it is better to git the whole data to the front to deal with it too
so the res should look like that 
```
{
    "average_progress": 0.5,
    "tasks_count": 10,
    "tasks_completed_count": 5
}
```

Reply:
Done